DATASETS_v2.md

```diff
@@ -0,0 +1,244 @@
+Datasets V2
+-----------
+
+### Background 
+
+Sno 0.2 introduced Datasets V1, and Sno 0.5 introduced Datasets V2.
+Datasets V1 and V2 are storage formats for database tables where each table row is stored in a separate file.
+This means table rows can be stored using git-style version control, resulting in a storage format for a database-table which has version history.
+
+Datasets V2 is very similar to Datasets V1 - the main difference is that the schema of a Datasets V2 table can be changed in isolation without having
+to rewrite every row in the table. Rows that were written with a previous schema are adapted to fit the current schema when read.
+
+The following is a technical description of datasets V2.
+
+### Overall structure
+
+A V2 dataset is a folder named `.sno-dataset` that contains two folders. These are `meta` which contains information about the database table itself - its title, description, and schema - and 'feature' which contains the table rows. The schema file contains the structure of the table rows, ie the names and type of each column. The "name" of the V2 Dataset is the path to the `.sno-dataset` folder. 
+
+For example, here is the basic folder structure of a dataset named contours/500m:
+
+```
+contours/
+contours/500m/
+contours/500m/.sno-dataset/
+contours/500m/.sno-dataset/meta/
+contours/500m/.sno-dataset/meta/title              # Title of the dataset
+contours/500m/.sno-dataset/meta/description        # Description of the dataset
+contours/500m/.sno-dataset/meta/schema.json        # Schema of the dataset 
+contours/500m/.sno-dataset/meta/...                # Other dataset metadata
+
+contours/500m/.sno-dataset/feature/...             # Database table rows
+```
+
+### Meta items
+
+The following items are stored in the meta part of the dataset, and have the following structure.
+
+#### `meta/title` 
+Contains the title of the dataset, encoded using UTF-8. The title is freeform text, clients could if they desired include HTML tags or markdown in the title.
+
+#### `meta/description`
+Contains the title of the dataset, encoded using UTF-8. The description is freeform text, clients could if they desired include HTML tags or markdown in the title.
+
+#### `meta/schema.json`
+Contains the current schema of the table, as a JSON array. Each item in the array represents a column in the table, and contains the name and type of that database column. A more in-depth explanation is found below under the heading "Syntax".
+
+For example, here is the schema for a dataset containing building outlines:
+```json
+[
+  {
+    "id": "500f5ecc-c02c-e8db-052a-84efb4d08a00",
```

Comment:
No... in python I keep construct these in column order (ie, first column first), which python3 preserves, so that if I output them again they remain in column order for readability reasons. 

Here in the documentation I could sort or shuffle them without it making the documentation incorrect, but, seems best to keep them in column order still since they *are* in that order in the code and in the database row - putting them in some other order won't make it clearer.